modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/StateChannelSnapshotBinary.scala

```diff
@@ -14,10 +15,12 @@ case class StateChannelSnapshotBinary(
   lastSnapshotHash: Hash,
   content: Array[Byte]
 )
-
 object StateChannelSnapshotBinary {
   implicit val show: Show[StateChannelSnapshotBinary] = s => s"StateChannelSnapshotBinary(lastSnapshotHash=${s.lastSnapshotHash.show})"
 
+  implicit val arrayByteOrder: Order[Array[Byte]] = Order.by(_.toSeq)
+  implicit val order: Order[StateChannelSnapshotBinary] = Order.whenEqual(Order.by(_.lastSnapshotHash), Order.by(_.content))
```

Comment:
I think that this can be derived atuomatically by derevo once we have an order for `Array`

Reply:
yup, that's true